bundles/org.openhab.binding.chatgpt/src/main/resources/OH-INF/addon/addon.xml

```diff
@@ -8,4 +8,8 @@
 	<description>This binding allows interaction with OpenAI's ChatGPT.</description>
 	<connection>cloud</connection>
 
+	<service-id>org.openhab.voice.chatgpthli</service-id>
```

Comment:
According to the XSD, the service-id is the primary service for the add-on - which should usually be the `ChatGPTHandlerFactory`.

Thinking further about it, we actually might want individual configurations for each Account thing / HLI service and not the same for all.
Might it make sense to put the configuration directly onto the Account thing? The HLI has access to the ThingHandler and thus could read if from there. Wdyt?

Reply:
You said that the HLI service should not touch anything around the channels, so I made the configuration for the HLI separately in the add-on configuration page. For other account things, you can configure it the same way as before - in the channel configuration via account thing, each thing with individual config. I'm a little confused and don't understand how this should be correct? And why can't it be like it is now?

Reply:
With the current implementation, the configuration turned up under "Add-ons" -> "ChatGPT". The problem is that an add-on can only provide a single configurable service. If you e.g. implement the Whisper-TTS add-on, this would be a "voice" add-on and it is clear that the configuration is (only) voice related.
But in our case here, the "main" feature is the binding and the HLI service is just some additional part to it. As there can be only a single configurable service, it must be one that is there for "all features" of the add-on - that's why I believe it should be the handler factory service. As we have different features in the add-on, our configuration could have multiple configuration parameter groups - one for the binding (currently empty) and one for the HLI. Nonetheless, the factory would need to pass on the configuration to the different parts of the add-on.
I agree, this isn't overly elegant, but I don't see a way to easily lift the constraint of having a single configurable service per add-on.